test/statistics_test.py

```diff
@@ -0,0 +1,161 @@
+# Copyright (C) 2022 by Daniel Shapero <shapero@uw.edu>
+#
+# This file is part of icepack.
+#
+# icepack is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# The full text of the license can be found in the file LICENSE in the
+# icepack source directory or at <http://www.gnu.org/licenses/>.
+
+import pytest
+import numpy as np
+import numpy.random as random
+import firedrake
+from firedrake import inner, grad, dx, exp, interpolate, as_vector
+import icepack
+from icepack.statistics import StatisticsProblem, MaximumProbabilityEstimator
+from icepack.constants import (
+    ice_density as ρ_I,
+    water_density as ρ_W,
+    gravity as g,
+    glen_flow_law as n,
+)
+
+
+def test_poisson_problem():
+    Nx, Ny = 32, 32
+
+    mesh = firedrake.UnitSquareMesh(Nx, Ny)
+    degree = 2
+    Q = firedrake.FunctionSpace(mesh, "CG", degree)
+    V = firedrake.FunctionSpace(mesh, "CG", degree)
+
+    x, y = firedrake.SpatialCoordinate(mesh)
+    f = interpolate(firedrake.Constant(1), Q)
+
+    def simulation(q):
+        u = firedrake.Function(V)
+        J = (0.5 * exp(q) * inner(grad(u), grad(u)) - f * u) * dx
+        F = firedrake.derivative(J, u)
```

Comment:
I think it's a little confusing to call this `J` and then take the derivative to get the weak form of `F` since `J` usually refers to the functional we wish to reduce. Can you rename `J` or explain why I'm wrong? I personally prefer to just explicitly put the `TestFunction`s in to `F`

Reply:
Ooh great point! So the reason I did this is that, under the hood, the momentum conservation equation in icepack is formulated not as the weak form of a PDE but as a minimization principle, from which one can then derive the weak form. I wanted to test the stats stuff on a linear problem first and, for consistency, I used the minimization form of the Poisson equation. So there's an inner minimization problem (to solve the model physics) and an outer one (estimate the parameters `q`) and it is admittedly confusing to use the same letter for both. I'll find a better one, open to suggestions.

Reply:
Well if you want to maintain your consistency you can always just add a comment in case someone else like me comes along and thinks it's odd

Reply:
Also I've never thought about solving the PDE as a minimization problem before - that's pretty neat